File: frontend/src/api/jd.ts
/**
 * 文件名: api/jd.ts
 * 创建时间: 2026-06-26
 * 作者: TalentSense Team
 * 功能描述: JD 匹配接口
 */
import request from './request'
import type { JdMatchResult } from '@/types/jd'

/**
 * JD 文本匹配候选人
 * @param jdText JD 文本
 * @param topK 返回数量
 */
export function matchJd(jdText: string, topK: number): Promise<JdMatchResult> {
  return request.post('/jd/match', { jd_text: jdText, top_k: topK }) as unknown as Promise<JdMatchResult>
}

File: backend/app/models/jd.py
"""
文件名: app/models/jd.py
创建时间: 2026-06-26
作者: TalentSense Team
功能描述: JD 匹配模型，对应 API-Design.md 七、JD Match
"""
from pydantic import BaseModel


class JdMatchRequest(BaseModel):
    jd_text: str
    top_k: int = 10
    filters: dict = {}


class ParsedRequirements(BaseModel):
    skills: list[str] = []
    work_years_min: int | None = None
    education_min: int | None = None
    responsibilities: list[str] = []


class MatchAnalysis(BaseModel):
    matched_skills: list[str]
    missing_skills: list[str]
    experience_match: bool
    education_match: bool
    reason: str
